### PR TITLE
Adds proper sorting for all index pages

### PR DESCRIPTION
**Pull request by ghatighorias**
Sorting is the following for index pages.

For user related controllers (student, teacher, volunteer, coordinator and user controllers)
- Sorting is ascending based on Name, family_name and nickname

For Entities with Date or time (task, event, attendance, class, offered_course, term)
- Sorting is descending based on the following order [date, start time, finishing time]

For Course since it only has a name
- Sorting is ascending based on its name

### CODE REVIEW

File: lib/course_planner_web/controllers/term_controller.ex [hunk 1]
```diff
@@ -2,14 +2,17 @@ defmodule CoursePlannerWeb.TermController do
...
+  alias CoursePlanner.{Terms, Terms.Term}
   alias Ecto.Changeset
 
   import Canary.Plugs
   plug :authorize_controller
 
   def index(conn, _params) do
-    render(conn, "index.html", terms: Terms.all)
+    query = from t in Term, order_by: [desc: t.start_date, desc: t.end_date]
+    terms = Repo.all(query)
```
Comment (rhnonose) [inline lib/course_planner_web/controllers/term_controller.ex]: Can you put this inside `terms.ex`?

File: lib/course_planner_web/controllers/course_controller.ex [hunk 2]
```diff
@@ -8,7 +8,9 @@ defmodule CoursePlannerWeb.CourseController do
   plug :authorize_controller
 
   def index(conn, _params) do
-    courses = Repo.all(Course)
+    query = from c in Course, order_by: [asc: c.name]
+    courses = Repo.all(query)
```
Comment (rhnonose) [inline lib/course_planner_web/controllers/course_controller.ex]: Can you put this inside `courses.ex`?

File: lib/course_planner_web/controllers/class_controller.ex [hunk 3]
```diff
@@ -13,7 +13,8 @@ defmodule CoursePlannerWeb.ClassController do
     join: co in assoc(oc, :course),
     join: c in assoc(oc, :classes),
     preload: [offered_courses: {oc, classes: c, course: co}],
-    order_by: [asc: t.start_date, asc: co.name, asc: c.date, asc: c.starting_at, asc: c.finishes_at]
+    order_by: [desc: t.start_date, desc: co.name, desc: c.date,
+               desc: c.starting_at, desc: c.finishes_at]
```
Comment (rhnonose) [inline lib/course_planner_web/controllers/class_controller.ex]: Can you move this query inside `classes.ex`?

File: lib/course_planner_web/controllers/course_controller.ex (hunk 2 above)
Comment (ghatighorias) [reply lib/course_planner_web/controllers/course_controller.ex]: done

File: lib/course_planner_web/controllers/term_controller.ex (hunk 1 above)
Comment (ghatighorias) [reply lib/course_planner_web/controllers/term_controller.ex]: done

File: lib/course_planner_web/controllers/class_controller.ex (hunk 3 above)
Comment (ghatighorias) [reply lib/course_planner_web/controllers/class_controller.ex]: done

**Pull request merged**